src/main/java/org/jabref/logic/bibtex/comparator/BibDatabaseDiff.java

```diff
@@ -83,28 +69,26 @@ private static List<BibEntryDiff> compareEntries(List<BibEntry> originalEntries,
         }
 
         // Now we've found all exact matches, look through the remaining entries, looking for close matches.
-        for (Iterator<BibEntry> iteratorNotMatched = notMatched.iterator(); iteratorNotMatched.hasNext(); ) {
-            BibEntry originalEntry = iteratorNotMatched.next();
-
+        DuplicateCheck duplicateCheck = new DuplicateCheck(new BibEntryTypesManager());
```

Comment:
The BibEntryTypesManager contains sets with all entry types and their fields. I really would avoid creating this class every time. Ideally you pass the reference down or at least make it an instance variable

Reply:
Decided to do this in a Follow up pr